### DISCUSSION TITLE
Understanding how aiProcess_JoinIdenticalVertices works

### DISCUSSION THREAD

**Discussion by NicusorN5** [assimp/assimp#5160] [Q&A] [answered]
Hi, I'm trying to load a model using Assimp:

Using `aiProcess_JoinIdenticalVertices` removes a vertex that is required for the first submesh. It seems to remove a needed vertex in the first submesh. When using `aiProcess_JoinIdenticalVertices` I expected the meshes to have non repeating vertices per vertex buffer,, but I didn't expect all the meshes to have unique vertices in the entire model. 

My model is made from two meshes.
The first mesh has a missing triangle and the second mesh is perfectly correct in either case.

Picture with the model using the `aiProcess_JoinIdenticalVertices` flag (wrong):
![image](https://github.com/assimp/assimp/assets/20599225/673ac92d-3bfc-46eb-a197-6ee0faa35a5e)

Picture without using the flag (correct):
![image](https://github.com/assimp/assimp/assets/20599225/6555901e-b67a-45eb-88f3-09ae9453d653)

Buffers when using the flag:
![image](https://github.com/assimp/assimp/assets/20599225/da73d962-a6b0-417a-84b2-e3347c216919)
![image](https://github.com/assimp/assimp/assets/20599225/7263b77a-7b75-4758-ab64-a1a0335b10da)

(I can't seem to copy the entire vertex data here, sorry)

Buffers when not using the flag:
![image](https://github.com/assimp/assimp/assets/20599225/eaef12d9-8c7d-4efb-831c-c47d0a038347)
![image](https://github.com/assimp/assimp/assets/20599225/0adb8967-9c5c-4bdb-9de7-920c376d989e)

When not using the flag, index buffers are not necessary, as expected.

If I did read the documentation right, using `aiProcess_JoinIdenticalVertices` is absolutely necessary if using index buffers.

My question is: How do I correctly generate the vertex and index buffer after `aiProcess_JoinIdenticalVertices` is set?

Importing source code is here : https://github.com/NicusorN5/3D_Rad_Space/blob/5f8bb4b5be81b7b369bfd2bafb57f4d8564048fd/3DRadSpace/Engine3DRadSpace/Graphics/Model3D.cpp#L31C2-L31C2

Repository: https://github.com/NicusorN5/3D_Rad_Space

Model (.X format) :  [Camera.zip](https://github.com/assimp/assimp/files/11858316/Camera.zip)

Any help is greatly appreciated, thanks!

**Comment by Rynnya** [comment, marked as answer]
Buffers are created correctly, you can see there's 6 indices and 4 vertices
Your issue is that you call `Draw` instead of `DrawIndexed` in your [DrawVertexBufferWithIndices](https://github.com/NicusorN5/3D_Rad_Space/blob/02629b4cd7ddf3b6c6f5225c88b517db55c3d138/3DRadSpace/Engine3DRadSpace/GraphicsDevice.cpp#L146C52-L146C52)
I'm not sure how D3D11 driver handles such behaviour when he expect 3 vertices, but I guess it just silently ignores additional vert
It's NOT Assimp issue, it's your mistake that you must fix

**Reply by NicusorN5** [reply]
Hi, re-enabling the flag and Using `ID3D11Context::DrawIndexed` for some reason the winding order was reversed, so I had to use the `aiProcess_FlipWindingOrder` importer flag. I'm not sure if this is recommended, but everything workes perfectly now.

Thanks for the heads-up.
